Code/GraphMol/MolStandardize/Wrap/testMolStandardize.py

```diff
@@ -16,207 +16,238 @@
+
+  def test11FragmentParams(self):
+    data = """//   Name	SMARTS
 fluorine	[F]
 chlorine	[Cl]
         """
-        fragremover = rdMolStandardize.FragmentRemoverFromData(data)
-        mol = Chem.MolFromSmiles("CN(C)C.Cl.Cl.Br")
-        nm = fragremover.remove(mol)
-        self.assertEqual(Chem.MolToSmiles(nm), "Br.CN(C)C")
+    fragremover = rdMolStandardize.FragmentRemoverFromData(data)
+    mol = Chem.MolFromSmiles("CN(C)C.Cl.Cl.Br")
+    nm = fragremover.remove(mol)
+    self.assertEqual(Chem.MolToSmiles(nm), "Br.CN(C)C")
 
-    def test12ChargeParams(self):
-        params = """// The default list of AcidBasePairs, sorted from strongest to weakest.
+  def test12ChargeParams(self):
+    params = """// The default list of AcidBasePairs, sorted from strongest to weakest.
 // This list is derived from the Food and Drug: Administration Substance
 // Registration System Standard Operating Procedure guide.
 //
 //	Name	Acid	Base
 -SO2H	[!O][SD3](=O)[OH]	[!O][SD3](=O)[O-]
 -SO3H	[!O]S(=O)(=O)[OH]	[!O]S(=O)(=O)[O-]
 """
-        mol = Chem.MolFromSmiles("C1=C(C=CC(=C1)[S]([O-])=O)[S](O)(=O)=O")
-        # instantiate with default acid base pair library
-        reionizer = rdMolStandardize.ReionizerFromData(params, [])
-        print("done")
-        nm = reionizer.reionize(mol)
-        self.assertEqual(Chem.MolToSmiles(nm), "O=S([O-])c1ccc(S(=O)(=O)O)cc1")
-
+    mol = Chem.MolFromSmiles("C1=C(C=CC(=C1)[S]([O-])=O)[S](O)(=O)=O")
+    # instantiate with default acid base pair library
+    reionizer = rdMolStandardize.ReionizerFromData(params, [])
+    print("done")
+    nm = reionizer.reionize(mol)
+    self.assertEqual(Chem.MolToSmiles(nm), "O=S([O-])c1ccc(S(=O)(=O)O)cc1")
+
+  def test13Tautomers(self):
+    enumerator = rdMolStandardize.TautomerEnumerator()
+    m = Chem.MolFromSmiles("C1(=CCCCC1)O")
+    ctaut = enumerator.Canonicalize(m)
+    self.assertEqual(Chem.MolToSmiles(ctaut), "O=C1CCCCC1")
+
+    params = rdMolStandardize.CleanupParameters()
+    enumerator = rdMolStandardize.TautomerEnumerator(params)
+    m = Chem.MolFromSmiles("C1(=CCCCC1)O")
+    ctaut = enumerator.Canonicalize(m)
+    self.assertEqual(Chem.MolToSmiles(ctaut), "O=C1CCCCC1")
+
+    tauts = enumerator.Enumerate(m)
+    self.assertEqual(len(tauts), 2)
+    ctauts = list(sorted(Chem.MolToSmiles(x) for x in tauts))
+    self.assertEqual(ctauts, ['O=C1CCCCC1', 'OC1=CCCCC1'])
+
+    def scorefunc1(mol):
+      ' stupid tautomer scoring function '
+      p = Chem.MolFromSmarts('[OH]')
+      return len(mol.GetSubstructMatches(p))
+
+    def scorefunc2(mol):
+      ' stupid tautomer scoring function '
+      p = Chem.MolFromSmarts('O=C')
+      return len(mol.GetSubstructMatches(p))
+
+    m = Chem.MolFromSmiles("C1(=CCCCC1)O")
+    ctaut = enumerator.Canonicalize(m, scorefunc1)
+    self.assertEqual(Chem.MolToSmiles(ctaut), "OC1=CCCCC1")
+    ctaut = enumerator.Canonicalize(m, scorefunc2)
+    self.assertEqual(Chem.MolToSmiles(ctaut), "O=C1CCCCC1")
```

Comment:
Might be worth writing a function with the wrong API to see what happens :)

Reply:
good point. I also added one to make sure/demonstrate that you can use lambdas from Python
(boost.python is absolutely magic)

Reply:
Yeah, this aspect of boost is pretty awesome.